operator-framework-core/src/test/java/io/javaoperatorsdk/operator/OperatorTest.java

```diff
@@ -0,0 +1,75 @@
+package io.javaoperatorsdk.operator;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.api.config.ConfigurationService;
+import io.javaoperatorsdk.operator.api.config.ControllerConfiguration;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class OperatorTest {
+
+  private final KubernetesClient kubernetesClient = mock(KubernetesClient.class);
+  private final ConfigurationService configurationService = mock(ConfigurationService.class);
+  private final ControllerConfiguration configuration = mock(ControllerConfiguration.class);
+
+  private final Operator operator = new Operator(kubernetesClient, configurationService);
+  private final FooReconciler fooReconciler = FooReconciler.create();
+
+  @Test
+  @DisplayName("should register `Reconciler` to Controller")
+  public void shouldRegisterReconcilerToController() {
+    // given
+    when(configurationService.getConfigurationFor(fooReconciler)).thenReturn(configuration);
+    when(configuration.watchAllNamespaces()).thenReturn(true);
+    when(configuration.getName()).thenReturn("FOO");
+    when(configuration.getCustomResourceClass()).thenReturn(FooReconciler.class);
+
+    // when
+    operator.register(fooReconciler);
+
+    // then
+    verify(configuration).watchAllNamespaces();
```

Comment:
Not sure I like this test as this is more testing the internal implementation than checking that the reconciler is actually registered.

Reply:
@metacosm Thanks for your advice!  I added a test code to check if the controller was added normally. :) 

```
    assertThat(operator.getControllers().size()).isEqualTo(1);
    assertThat(operator.getControllers().get(0).getReconciler()).isEqualTo(fooReconciler);
```